maven-core/src/main/java/org/apache/maven/session/scope/internal/SessionScope.java

```diff
@@ -19,35 +19,23 @@
  * under the License.
  */
 
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.LinkedList;
+import java.util.Collection;
+import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
 
 import com.google.inject.Key;
 import com.google.inject.OutOfScopeException;
 import com.google.inject.Provider;
 import com.google.inject.Scope;
-import com.google.inject.util.Providers;
 
 /**
  * SessionScope
  */
 public class SessionScope
-    implements Scope
+        implements Scope
 {
-    /**
-     * @since 3.3.0
-     */
-    public static class Memento
```

Comment:
Same here

Reply:
Same answer.  SessionScope is internal, so I don't think it needs a deprecation notice.